ANSWERS.md

```diff
@@ -1,4 +1,20 @@
-- [ ] Why would you use class component over function components (removing hooks from the question)?
-- [ ] Name three lifecycle methods and their purposes.
-- [ ] What is the purpose of a custom hook?
-- [ ] Why is it important to test our apps?
\ No newline at end of file
+1. Why would you use class component over function components (removing hooks from the question)?
+
+If you need to use lifecycle hooks, you must use class components. If you need state in your component, it’s also necessary to use class components.
+
+2. Name three lifecycle methods and their purposes.
+
+Mounting - putting elements into the DOM (example method is “constructor”)
+
+Updating - A component is updated whenever there is a change in the component's state or props. (Example method is “getDerivedStateFromProps”)
```

Comment:
getDerivedStateFromProps is a method that doesn't get talked about too often. 👍 awesome